spec/lib/wolf_waagen_api/import_spec.rb

```diff
@@ -0,0 +1,250 @@
+      report: other_report,
+      address: 4
+    )
+  end
+
+  describe '#initialize' do
+    it 'validates report' do
+      expect do
+        described_class.new(report: nil)
+      end.to raise_error ArgumentError
+    end
+
+    it 'ensures report has a `hive_id`' do
+      expect do
+        described_class.new(report: Report.first)
+      end.to raise_error ArgumentError
+    end
+  end
+
+  describe '#api_sensors' do
+    subject do
+      described_class.new(report: report).api_sensors
+    end
+
+    context 'given sensors of types not sepcified in constant' do
+      before { other_type_sensor }
+
+      it {
+        is_expected.not_to include(other_type_sensor)
+      }
+    end
+
+    context 'given sensor for other reports' do
+      before { other_type_sensor }
+
+      it { is_expected.not_to include(other_report_sensor) }
+    end
+
+    context 'given no matching sensor' do
+      before { other_sensor_type }
+
+      it { is_expected.to eq([]) }
+    end
+
+    context 'given no sensor types match' do
+      it { is_expected.to eq([]) }
+    end
+  end
+
+  describe '#api_sensors_by_series_id' do
+    subject do
+      specified_type = WolfWaagenApi::Import::SENSOR_TYPES.first
+      import = described_class.new(report: report)
+      import.api_sensors_by_series_id(series_id: specified_type[:series_id])
+    end
+
+    context 'if no sensor type matches' do
+      before do
+        other_type_sensor
+      end
+
+      it { is_expected.to eq([]) }
+    end
+
+    context 'if no sensor matches' do
+      before do
+        wolf_sensor_type
+      end
+
+      it { is_expected.to eq([]) }
+    end
+  end
+
+  describe '#start_date' do
+    subject { described_class.new(report: report).start_date }
+
+    context 'given no sensor readings for one of two sensors' do
+      before do
+        wolf_sensor
+        wolf_sensor2
+        wolf_sensor_reading
+      end
+
+      it 'returns the report start date' do
+        is_expected.to eq(DateTime.new(2019, 1, 1))
+      end
+    end
+
+    context 'given sensor readings for all sensors' do
+      before do
+        wolf_sensor
+        wolf_sensor2
+        wolf_sensor_reading
+        wolf_sensor_reading2
+      end
+
+      it 'returns the oldest latest sensor reading date' do
+        is_expected.to eq(DateTime.new(2019, 1, 2))
+      end
+    end
+  end
+
+  describe '#save_series_points' do
+    let(:request) do
+      WolfWaagenApi::Request.new(
+        hive_id: 'HIVE_ID',
+        start_date: DateTime.new(2019, 1, 1, 0, 0),
+        end_date: DateTime.new(2019, 1, 2, 0, 0)
+      )
+    end
+
+    let(:result) do
+      WolfWaagenApi::Result.new(
+        request: request,
+        data: {
+          pointStart: 1546300800000, # 2019-01-01 00:00:00
+          pointInterval: 60 * 60 * 1000,
+          series: [{
+            id: WolfWaagenApi::Import::SENSOR_TYPES.first[:series_id],
+            values: [10, 20, 30]
+          }]
+        }
+      )
```

Comment:
The need to setup a lot of dependencies to test a small class is a strong indicator for a code-smell and probably a violation against the law of demeter. https://en.wikipedia.org/wiki/Law_of_Demeter